lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/time/runtime/RuntimeClock.java

```diff
@@ -213,66 +272,73 @@ public void run() {
       this.handlers.onNext(new RuntimeStart(this.timer.getCurrent()));
 
       LOG.log(Level.FINE, "Initiate start time");
-      final StartTime start = new StartTime(this.timer.getCurrent());
-      this.handlers.onNext(start);
+      this.handlers.onNext(new StartTime(this.timer.getCurrent()));
+
+      for (;;) {
```

Comment:
`while (true)` seems more idiomatic to me. What do we gain by this `for` construction?